src/python/gudhi/point_cloud/knn.py

```diff
@@ -0,0 +1,317 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Marc Glisse
+#
+# Copyright (C) 2020 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+import numpy
+
+# TODO: https://github.com/facebookresearch/faiss
+
+
+class KNearestNeighbors:
+    """
+    Class wrapping several implementations for computing the k nearest neighbors in a point set.
+    """
+
+    def __init__(self, k, return_index=True, return_distance=False, metric="euclidean", **kwargs):
+        """
+        Args:
+            k (int): number of neighbors (possibly including the point itself).
+            return_index (bool): if True, return the index of each neighbor.
+            return_distance (bool): if True, return the distance to each neighbor.
+            implementation (str): choice of the library that does the real work.
+
+                * 'keops' for a brute-force, CUDA implementation through pykeops. Useful when the dimension becomes large (10+) but the number of points remains low (less than a million). Only "minkowski" and its aliases are supported.
```

Comment:
Some documentation are really long lines. Please let lines <= 120.
Here I think you can jump to the next line with 4 spaces at the beginning, it shall work.

Reply:
I am quite sure I tried that already and it doesn't work, not for a list nested in a param. That's why I kept those very long lines. Théo hit a similar issue IIRC. If you can find a way to split the lines without breaking the rendering by sphinx, I'd be happy to split those lines indeed.